lib/types/object/index.js

```diff
@@ -118,16 +118,31 @@ internals.Object = class extends Any {
                 }
             }
 
-            if (target[rename.from] === undefined) {
+
+
+            if (target[rename.from] === undefined && !(rename.from instanceof RegExp)) {
```

Comment:
We call `rename.from instanceof RegExp` 3 times in here seems like a good case to use a boolean and just reference that. ie. `let isRegExRename = rename.from instanceof RegExp`